lib/libvcc/vcc_expr.c

```diff
@@ -189,7 +190,17 @@ vcc_expr_edit(struct vcc *tl, vcc_type_t fmt, const char *p, struct expr *e1,
 			    "VPI_BundleStrands(%d, &strs_%u_a, strs_%u_s,"
 			    "\v+\n%s,\nvrt_magic_string_end\v-\n)",
 			    e3->nstr, tl->unique, tl->unique,
-			VSB_data(e3->vsb));
+			    VSB_data(e3->vsb));
+			tl->unique++;
+			break;
+		case 'B':
+			AZ(e2);
+			assert(e1->fmt == STRING);
+			VSB_printf(tl->curproc->prologue,
+			    "  struct vrt_blob blobs_%u;\n", tl->unique);
```

Comment:
we cannot put blobs on the stack, this would break out contract:
https://github.com/varnishcache/varnish-cache/blob/fe687fbe9b9514e7c3a7d0d332404e8538d5a804/include/vrt.h#L191

Reply:
Good catch, that should be easy to fix: pass a `VRT_CTX` to `VPI_BLOB` instead of the on-stack blob placeholder. Will do.

Reply:
why not just use `VRT_blob()` ?

Reply:
Because `VPI_Blob` handles the null case, and of course it can then call `VRT_blob`. I can't simply call it with `strlen()` for the length argument.

Reply:
764207b325094a89d53af135d887ebfda41a17d8